src/routes/transactions/mappers/multisig-transaction.mapper.ts

```diff
@@ -0,0 +1,592 @@
+    return this.ERC20_TRANSFER_METHODS.some(
+      (method) => method === dataDecoded?.method,
+    );
+  }
+
+  private isErc721Transfer(transaction: MultisigTransaction): boolean {
+    const { dataDecoded } = transaction;
+    return this.ERC721_TRANSFER_METHODS.some(
+      (method) => method === dataDecoded?.method,
+    );
+  }
+
+  private getFromParam(dataDecoded: DataDecoded, fallback: string): string {
+    if (!dataDecoded.parameters) {
+      return fallback;
+    }
+
+    switch (dataDecoded.method) {
+      case this.TRANSFER_FROM_METHOD:
+      case this.SAFE_TRANSFER_FROM_METHOD:
+        return typeof dataDecoded.parameters[0]?.value === 'string'
+          ? dataDecoded.parameters[0]?.value
+          : fallback;
+      case this.TRANSFER_METHOD:
+      default:
+        return fallback;
+    }
+  }
+
+  private getToParam(dataDecoded: DataDecoded, fallback: string): string {
+    if (!dataDecoded?.parameters) {
+      return fallback;
+    }
+
+    switch (dataDecoded.method) {
+      case this.TRANSFER_METHOD:
+        return typeof dataDecoded.parameters[0]?.value === 'string'
+          ? dataDecoded.parameters[0]?.value
+          : fallback;
+      case this.TRANSFER_FROM_METHOD:
+      case this.SAFE_TRANSFER_FROM_METHOD:
+        return typeof dataDecoded.parameters[1]?.value === 'string'
+          ? dataDecoded.parameters[1]?.value
+          : fallback;
+      default:
+        return fallback;
+    }
+  }
+
+  private checkSenderOrReceiver(transaction: MultisigTransaction): boolean {
+    const { dataDecoded } = transaction;
+    if (!dataDecoded) return false;
+    return (
+      this.TRANSFER_METHOD == dataDecoded.method ||
+      this.getFromParam(dataDecoded, '') === transaction.safe ||
+      this.getToParam(dataDecoded, '') === transaction.safe
+    );
+  }
+
+  private isCustomTransaction(
+    value: number,
+    dataSize: number,
+    operation: Operation,
+  ): boolean {
+    return (value > 0 && dataSize > 0) || operation !== 0;
+  }
+
+  private isEtherTransfer(value: number, dataSize: number): boolean {
+    return value > 0 && dataSize === 0;
+  }
+
+  private isSettingsChange(
+    transaction: MultisigTransaction,
+    value: number,
+    dataSize: number,
+  ): boolean {
+    return (
+      value === 0 &&
+      dataSize > 0 &&
+      transaction.safe === transaction.to &&
+      this.SETTINGS_CHANGE_METHODS.includes(transaction.dataDecoded?.method)
+    );
+  }
+
+  private isValidTokenTransfer(transaction: MultisigTransaction): boolean {
+    return (
+      (this.isErc20Transfer(transaction) ||
+        this.isErc721Transfer(transaction)) &&
+      this.checkSenderOrReceiver(transaction)
+    );
+  }
+
+  private async mapTransactionInfo(
+    chainId: string,
+    transaction: MultisigTransaction,
+    safe: Safe,
+  ): Promise<TransactionInfo> {
+    const value = Number(transaction?.value) || 0;
+    const dataSize = transaction.data
+      ? (Buffer.byteLength(transaction.data) - 2) / 2
```

Comment:
Do you mean if `Buffer.byteLength(transaction.data)` is `0`? 

`transaction.data` is a string

Reply:
In that case `transaction.data` would be a falsy value so the ternary operation would return `0`. Maybe there's a corner case if `transaction.data` is `1`, because `dataSize` becomes negative in that case, but reviewing the code it seems it won't cause an error either. I don't have all the context here, I'm just trying to mimic the Rust implementation, do you have any concern about it?

Reply:
Ok I see what you mean with `transaction.data` being `0`. In my opinion I think it's harder to read if we treat `""` as being the same as `undefined`.

Probably something like the following what do you think?

```typescript
const dataByteLength = transaction.data
  ? Buffer.byteLength(transaction.data)
  : 0;

const dataSize = dataByteLength >= 2 ? (dataByteLength - 2) / 2 : 0;
```

Reply:
Yes, good catch. I changed it in [912132b](https://github.com/5afe/safe-client-gateway-nest/pull/197/commits/912132beed905de946a60c305496cd05313a51ba), thanks!